heron/healthmgr/src/java/org/apache/heron/healthmgr/common/PhysicalPlanProvider.java

```diff
@@ -19,70 +19,129 @@
 
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
 import java.util.logging.Logger;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Provider;
 
-import com.microsoft.dhalion.events.EventHandler;
-import com.microsoft.dhalion.events.EventManager;
-
-import org.apache.heron.healthmgr.common.HealthManagerEvents.ContainerRestart;
-import org.apache.heron.healthmgr.common.HealthManagerEvents.TopologyUpdate;
+import org.apache.heron.api.generated.TopologyAPI;
 import org.apache.heron.proto.system.PhysicalPlans.PhysicalPlan;
+import org.apache.heron.proto.tmaster.TopologyMaster;
 import org.apache.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import org.apache.heron.spi.utils.NetworkUtils;
 
 import static org.apache.heron.healthmgr.HealthPolicyConfig.CONF_TOPOLOGY_NAME;
 
 /**
- * A topology's physical plan may get updated after initial deployment. This provider is used to
- * fetch the latest version from the state manager and provide to any dependent components.
+ * A topology's physical plan may get updated at runtime. This provider is used to
+ * fetch the latest version from the tmaster and provide to any dependent components.
  */
 public class PhysicalPlanProvider implements Provider<PhysicalPlan> {
   private static final Logger LOG = Logger.getLogger(PhysicalPlanProvider.class.getName());
 
   private final SchedulerStateManagerAdaptor stateManagerAdaptor;
   private final String topologyName;
 
-  private PhysicalPlan physicalPlan;
+  private PhysicalPlan cachedPhysicalPlan = null;
 
   @Inject
   public PhysicalPlanProvider(SchedulerStateManagerAdaptor stateManagerAdaptor,
-      EventManager eventManager, @Named(CONF_TOPOLOGY_NAME) String topologyName) {
+                              @Named(CONF_TOPOLOGY_NAME) String topologyName) {
     this.stateManagerAdaptor = stateManagerAdaptor;
     this.topologyName = topologyName;
-    eventManager.addEventListener(TopologyUpdate.class, new EventHandler<TopologyUpdate>() {
-      /**
-       * Invalidates cached physical plan on receiving topology update notification
-       */
-      @Override
-      public synchronized void onEvent(TopologyUpdate event) {
-        LOG.info(
-            "Received topology update event, invalidating cached PhysicalPlan: " + event.type());
-        physicalPlan = null;
-      }
-    });
-    eventManager.addEventListener(ContainerRestart.class, new EventHandler<ContainerRestart>() {
-      /**
-       * Invalidates cached physical plan on receiving container restart notification
-       */
-      @Override
-      public synchronized void onEvent(ContainerRestart event) {
-        LOG.info("Received container restart event, invalidating cached PhysicalPlan: "
-            + event.type());
-        physicalPlan = null;
-      }
-    });
   }
 
   @Override
   public synchronized PhysicalPlan get() {
```

Comment:
rename to `getPhysicalPlan` is easier to understand

Reply:
restricted by parent interface